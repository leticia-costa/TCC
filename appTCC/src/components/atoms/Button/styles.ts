import { StyleSheet } from "react-native";
import colors from "../../../utils/colors";

export const styles = StyleSheet.create({
  button: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: "center",
    gap: 12,
    backgroundColor: colors.purple,
    padding: 10,
    borderRadius: 15,
  },
  buttonText: {
    color: colors.white,
    fontSize: 16,
  },
  icon: {
    color: colors.white,
    fontSize: 18,
  }
});
