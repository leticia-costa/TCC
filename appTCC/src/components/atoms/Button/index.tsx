import React from "react";
import {
  StyleProp,
  TouchableOpacity,
  TouchableOpacityProps,
  ViewStyle,
} from "react-native";
import { styles } from "./styles";
import Typography from "../Typography";
import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
import { IconProp } from "@fortawesome/fontawesome-svg-core";

interface ButtonProps extends TouchableOpacityProps {
  label: string;
  style?: StyleProp<ViewStyle>;
  icon?: IconProp;
}
const Button: React.FC<ButtonProps> = ({
  label,
  icon,
  style,
  ...otherProps
}) => {
  return (
    <TouchableOpacity style={[styles.button, style]} {...otherProps}>
      <Typography style={styles.buttonText}>{label}</Typography>
      {!!icon && <FontAwesomeIcon icon={icon} style={styles.icon} size={35}/>}
    </TouchableOpacity>
  );
};

export default Button;
