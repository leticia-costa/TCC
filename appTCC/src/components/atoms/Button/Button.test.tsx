import React from "react";
import renderer from "react-test-renderer";
import Button from ".";
import { faHeart } from "@fortawesome/free-solid-svg-icons";
jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: "mock-fontawesome-icon",
}));
it("renders correctly", () => {
  const tree = renderer
    .create(
      <Button
        label="Teste"
        icon={faHeart}
        onPress={() => console.log("teste")}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
