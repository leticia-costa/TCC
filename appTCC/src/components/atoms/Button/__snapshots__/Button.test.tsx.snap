// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#440A8E",
      "borderRadius": 15,
      "display": "flex",
      "flexDirection": "row",
      "gap": 12,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Arial",
          "fontSize": 14,
        },
        {
          "color": "#ecf0f1",
          "fontSize": 16,
        },
      ]
    }
  >
    Teste
  </Text>
  <mock-fontawesome-icon
    icon={
      {
        "icon": [
          512,
          512,
          [
            128153,
            128154,
            128155,
            128156,
            128420,
            129293,
            129294,
            129505,
            9829,
            10084,
            61578,
          ],
          "f004",
          "M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z",
        ],
        "iconName": "heart",
        "prefix": "fas",
      }
    }
    size={35}
    style={
      {
        "color": "#ecf0f1",
        "fontSize": 18,
      }
    }
  />
</View>
`;
